lib/gemstash/gem_source/private_source.rb

```diff
@@ -118,6 +129,29 @@ def fetch_gem(gem_full_name)
         halt 403, "That gem has been yanked" unless gem.properties[:indexed]
         gem
       end
+
+      def fetch_gem_versions(gem)
+        results = db["
+          SELECT rubygem.name,
+                 version.number, version.platform
+          FROM rubygems rubygem
+          JOIN versions version
+            ON version.rubygem_id = rubygem.id
+          WHERE rubygem.name = ?
+            AND version.indexed = ?", gem.to_a, true].to_a
+        results.group_by {|r| r[:name] }.each do |gem, rows|
+          requirements = rows.group_by {|r| [r[:number], r[:platform]] }
+
+          value = requirements.map do |version, r|
```

Comment:
Would this do the right thing?

```suggestion
          value = requirements.map do |(version_number, platform), r|
```

Reply:
agreed. I changed it in the following commit. https://github.com/rubygems/gemstash/pull/277/commits/eb619cd6520c90b38a95ccb2960ad9e8d6de6d22